### PR TITLE
Made a few routines quicker

### PR DESCRIPTION
**Pull request by GabrielRavier**

### CODE REVIEW

**Comment by nemasu** [comment]
Hello!

Curious, how did you determine if the patches improve performance?

**Comment by GabrielRavier** [comment]
I just made a simple program that compared the speed of the routines ; my versions were ~40% quicker (mostly due to you pushing almost all registers at each call)

**Comment by nemasu** [comment]
Ah, I guess push/pop stack was a bad idea. I've been a bit busy lately, but I'll get around to this eventually.

File: string.asm [hunk 1]
```diff
@@ -94,13 +94,26 @@ string_atoi: ; rdi = string, rax = int
...
 
 string_copy: ; rdi = dest, rsi = source, rdx = bytes to copy
-    stackpush
-    mov rcx, rdx
-    inc rcx ; to get null
-    cld
-    rep movsb 
-    stackpop
-    ret
+    push rdx
```
Comment (triforce) [inline string.asm]: Indentation looks off here.

File: string.asm [hunk 2]
```diff
@@ -138,45 +151,54 @@ string_concat: ;rdi = string being added to, rsi = string to add, ret: new lengt
...
-    mov rax, r10
-    jmp string_contains_ret
-
-    string_contains_ret_no:
-    mov rax, -1
-
-    string_contains_ret:
-    stackpop
-    ret
+    push r9
```
Comment (triforce) [inline string.asm]: Indentation looks off here.

**Pull request closed without merging**